custom/logistics/api.py

```diff
@@ -250,30 +261,38 @@ def sms_user_sync(self, ilsgateway_smsuser, username_part=None, password=None,
                 user.save()
         return user
 
-    def add_language_to_user(self, logistics_sms_user, domains=None):
-        if not domains:
-            domains = []
+    def _reassign_number(self, user, phone_number):
+        v = VerifiedNumber.by_phone(phone_number, include_pending=True)
+        if v.domain in self._get_logistics_domains():
+            v.delete()
+            user.save_verified_number(self.domain, phone_number, True)
```

Comment:
Just realized, it could also be that the phone number format gets changed internally in the call to save() when calling apply_leniency (so that it won't find anything to delete in _reassign_number).  So you may also want to call apply_leniency early in the workflow (as soon as you get the phone number) and use the resulting phone number all throughout the rest of the workflow.